components/sup/src/manager/service/config.rs

```diff
@@ -43,6 +43,50 @@ static ENV_VAR_PREFIX: &'static str = "HAB";
 /// for a single service.
 static TOML_MAX_MERGE_DEPTH: u16 = 30;
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum UserConfigPath {
+    Recommended(PathBuf),
+    Deprecated(PathBuf),
+}
+
+impl UserConfigPath {
+    fn get_path(&self) -> &PathBuf {
+        match self {
+            &UserConfigPath::Recommended(ref p) |
+            &UserConfigPath::Deprecated(ref p) => p,
+        }
+    }
+
+    fn steal_path(self) -> PathBuf {
+        match self {
+            UserConfigPath::Recommended(p) |
+            UserConfigPath::Deprecated(p) => p,
+        }
+    }
+}
+
+pub trait PackageForConfig {
```

Comment:
I'm not sure the name of this trait accurately captures the intention behind the trait; what does `PackageForConfig` mean?

Reply:
I'll try to come up with some better name for it. And some docs. Probably will take me more time than addressing the rest of the issues here. ;)

This trait was meant to be basically a subset of the `Pkg` struct, so I could mock it in tests.